### ISSUE TITLE
Override/Remove a previously defined route

### ISSUE THREAD

**Issue by tfcporciuncula** [perwendel/spark#356]
I know [#277](https://github.com/perwendel/spark/issues/277) is about this and says that [#231](https://github.com/perwendel/spark/issues/231) solved it. The thing is, [what was done in #231](https://github.com/perwendel/spark/commit/6b76b6aaa62195cee887150b90427d97181e1292) was undone [here](https://github.com/perwendel/spark/commit/0305785a28611a46db110f4a834d15d1de8ee323). I don't think the changes in `SparkBase` ever made to master (even though the remove methods are still in [`SimpleRouteMatcher`](https://github.com/perwendel/spark/blob/master/src/main/java/spark/route/SimpleRouteMatcher.java), called by nobody).

Am I missing something? Was it supported and it's not anymore? Are there any plans to support this?

Thanks!

**Comment by travisspencer** [comment]
Spark has various levels of APIs. To remove routes, you need to use the lowest level API. You do this by inheriting from SparkBase and doing your removal by calling the appropriate methods on the matcher. So, it isn't that the removal methods aren't being used by anybody, they are! They just aren't surfaced through the higher-level APIs.

IMO, the [changes to SparkBase][1] should have been left, but the community felt they were useless. (We use them every day!) In any case, you now have to copy/paste that into your class that inherits from SparkBase and call the lower-level API on SparkBase if you want to remove routes.

If you too have this use case, perhaps Per and the community will reconsider adding them to SparkBase.

[1]: https://github.com/perwendel/spark/commit/0305785a28611a46db110f4a834d15d1de8ee323

**Comment by tfcporciuncula** [comment]
I was just playing with Spark and wondered whether this was supported or not, I wouldn't really say I have this use case. I opened the issue because I got really confused when I read the older issues about this and also because I thought it would be fun to be able to override routes on the run.

Thank you for the explanation, I really appreciate it. I guess closing this issue would make sense now, and hopefully, since I've referenced here the other issues about this, no one will get confused as I did.
